tiledb/sm/serialization/array_schema.cc

```diff
@@ -668,6 +668,52 @@ shared_ptr<Domain> domain_from_capnp(
   return make_shared<Domain>(HERE(), cell_order, dims, tile_order);
 }
 
+void dimension_label_to_capnp(
+    const DimensionLabel& dimension_label,
+    capnp::DimensionLabel::Builder* dim_label_builder,
+    const bool client_side) {
+  dim_label_builder->setDimensionId(dimension_label.dimension_index());
+  dim_label_builder->setName(dimension_label.name());
+  dim_label_builder->setUri(dimension_label.uri().to_string());
+  dim_label_builder->setAttributeName(dimension_label.label_attr_name());
+  dim_label_builder->setOrder(data_order_str(dimension_label.label_order()));
+  dim_label_builder->setType(datatype_str(dimension_label.label_type()));
+  dim_label_builder->setCellValNum(dimension_label.label_cell_val_num());
+  dim_label_builder->setExternal(dimension_label.is_external());
+  dim_label_builder->setRelative(dimension_label.uri_is_relative());
+
+  const auto& schema = dimension_label.schema();
+  auto schema_builder = dim_label_builder->initSchema();
+  throw_if_not_ok(array_schema_to_capnp(*schema, &schema_builder, client_side));
```

Comment:
The array schema is an optional element of the dimension label. It is a `nullptr` if the dimension label is part of a schema that was loaded from disk.

Reply:
So, I thought about it and I think the scenario that you are describing - null schema in dim label - would never occur in a Capnp serialization path, so even if the UT indeed fails, it's a kind of artificial scenario and we could not come up with an IT that fails as well. However, for safety I added a check to prevent that failure and make the UT pass as well, let me know if that seems good. (I had to do a minor fix to the UT as well).

Reply:
This looks good to me.